src/libs/ajax/loaded-data/useCachedData.ts

```diff
@@ -0,0 +1,31 @@
+import { Atom, LoadedState, ReadyState } from '@terra-ui-packages/core-utils';
+import { useLoadedDataEvents } from 'src/libs/ajax/loaded-data/useLoadedData';
+import { useSettableStore } from 'src/libs/react-utils';
+
+export const useCachedData = <S, HookResult extends LoadedState<S, unknown>>(
+  useHookInvoker: () => [HookResult, (dataCall: () => Promise<S>) => Promise<void>],
```

Comment:
Another naming nit: `useHookInvoker` sounds like there's an extra layer: like this is a hook that gets a hook invoker. Maybe `invokeHook`? That does lose the "use" hint that this needs to be called as a hook. But I think wrapping a hook inside another hook already looks kind of odd.

Reply:
went with useLoadedDataHook and improved the jsdocs

Reply:
I like it.